plugins/inventory/docker.py

```diff
@@ -0,0 +1,321 @@
+# Copyright (c) 2016, Paul Durivage <paul.durivage@gmail.com>
+# Copyright (c) 2016, Chris Houseknecht <house@redhat.com>
+# Copyright (c) 2016, James Tanner <jtanner@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+name: docker
+short_description: Ansible dynamic inventory plugin for Docker containers.
+version_added: 1.1.0
+author:
+    - Felix Fontein (@felixfontein)
+requirements:
+    - L(Docker SDK for Python,https://docker-py.readthedocs.io/en/stable/) >= 1.10.0
+extends_documentation_fragment:
+    - ansible.builtin.constructed
+    - community.docker.docker
+    - community.docker.docker.docker_py_1_documentation
+description:
+    - Reads inventories from the Docker API.
+    - Uses a YAML configuration file that ends with C(docker.[yml|yaml]).
+options:
+    plugin:
+        description:
+            - The name of this plugin, it should always be set to C(community.docker.docker)
+              for this plugin to recognize it as it's own.
+        type: str
+        required: true
+        choices: [ community.docker.docker ]
+
+    connection_type:
+        description:
+            - Which connection type to use the containers.
+            - Default is to use SSH (C(ssh)). For this, the options I(default_ip) and
+              I(private_ssh_port) are used.
+            - Alternatively, C(docker-cli) selects the
+              R(docker connection plugin,ansible_collections.community.docker.docker_connection),
+              and C(docker-api) selects the
+              R(docker_api connection plugin,ansible_collections.community.docker.docker_api_connection).
+        type: str
+        default: docker-api
+        choices:
+            - ssh
+            - docker-cli
+            - docker-api
+
+    verbose_output:
+        description:
+            - Toggle to (not) include all available inspection metadata.
+            - Note that all top-level keys will be transformed to the format C(docker_xxx).
+              For example, C(HostConfig) is converted to C(docker_hostconfig).
+            - If this is C(false), these values can only be used during I(constructed), I(groups), and I(keyed_groups).
+            - The C(docker) inventory script always added these variables, so for compatibility set this to C(true).
+        type: bool
+        default: false
+
+    default_ip:
+        description:
+            - The IP address to assign to ansible_host when the container's SSH port is mapped to interface
+              '0.0.0.0'.
+            - Only used if I(connection_type) is C(ssh).
+        type: str
+        default: 127.0.0.1
+
+    private_ssh_port:
+        description:
+            - The port containers use for SSH.
+            - Only used if I(connection_type) is C(ssh).
+        type: int
+        default: 22
+
+    add_legacy_groups:
```

Comment:
Why name it as a `legacy groups`? Why not `group_by` giving users a choice if they want to use particular groups. There we may add `legacy` as an option that will return output like the old script?

Reply:
Users can already create as many groups as they want in a very flexible way with `groups` and `keyed_groups`. I don't see why to add more choice. This here is mainly a convenience option to ease migration from the old inventory script.